### PR TITLE
fix: solve off by one error in solution

### LINKED ISSUE #42141: Josephus Problem - Off By One
**Affected page**
https://www.freecodecamp.org/learn/coding-interview-prep/rosetta-code/josephus-problem

**Your code**
```javascript
function josephus(init, kill) {
  // Setup
  let currPrisoner = 0;
  const prisoners = Array(init).fill(true);
  // Execute prisoners
  console.log("Kill Sequence:");
  for (let numAlive = init; numAlive > 1; numAlive--) {
    // Find next prisoner
    for (let numSkip = kill; numSkip > 0; numSkip--) {
      do {
        currPrisoner = (currPrisoner + 1) % init;
      } while (!prisoners[currPrisoner]);
    }
    // Execute prisoner
    console.log(currPrisoner);
    prisoners[currPrisoner] = false;
  }
  // Return last man standing
  console.log("Last Man Standing:");
  console.log(prisoners.indexOf(true));
  return prisoners.indexOf(true);
}

josephus(5, 2);
```

**Expected behavior**
The code that passes the test suite (given above) provides different output than the traditional description of the problem. I think the test suite has an 'off by one' error in initializing the starting position.

Current output with solution that passes test suite:
```
Kill Sequence:
2
4
1
0
Last Man Standing:
3
```
Expected output per problem description:
```
Kill Sequence:
1
3
0
4
Last Man Standing:
2
```
Relevant portion of problem description:
> For example, if there are n=5 prisoners and k=2, the order the prisoners are killed in (let's call it the "killing sequence") will be 1, 3, 0, and 4, and the survivor will be 2.

See also discussion here:
https://forum.freecodecamp.org/t/solution-to-rosetta-code-challenge-josephus-problem/460203

**Proposed Fix**
Change test suite to
> `josephus(30,3)` should return `28`

> `josephus(30,5)` should return `2`

> `josephus(20,2)` should return `8`

> `josephus(17,6)` should return `1`

> `josephus(29,4)` should return `1`

Bug reported by jsdisco on the forum.

### PR DESCRIPTION
**Pull request by alirezaghey**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #42141

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/josephus-problem.md [hunk 1]
```diff
@@ -86,36 +86,13 @@ function josephus(init, kill) {
...
-        current.prev.next = current.next;
-        current.next.prev = current.prev;
-        current = current.next;
-      }
-      return current.label;
-    }
+  const arr = Array.from(Array(init).keys());
+  let curr = -1
+  while (arr.length > 1) {
+    curr = (curr+kill)%arr.length;
```
Comment (gikf) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/josephus-problem.md:92]: Just a small nitpicking to put spaces around `+` and `%` here.

Comment (nhcarrigan) [reply curriculum/challenges/english/10-coding-interview-prep/rosetta-code/josephus-problem.md:92]:
```suggestion
    curr = (curr + kill) % arr.length;
```
A committable suggestion, as requested in our Discord Chat. @alirezaghey to commit this you can use the UI and click "Commit changes", create your commit message, and submit it. 🙂

**Review by gikf — APPROVED** [review]
LGTM :+1: 

Short, clean solution. Fixes the issue with inconsistent labelling. Tested against another solution, and compared removed elements on each step.

**Review by Sembauke — APPROVED** [review]
LGTM 👍

**Pull request merged**